### PR TITLE
[BACKLOG-37835] Add Databricks JDBC to Database UI

### PR DESCRIPTION
**Pull request by tgf**

### CODE REVIEW

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 56m 10s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl core,dbdialog,ui
```
---
:ok_hand: All tests passed!
 
Tests run: **5326**, Failures: **0**, Skipped: **14** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9117/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: dbdialog/src/main/java/org/pentaho/ui/database/event/DataHandler.java [hunk 1]
```diff
@@ -1456,6 +1469,8 @@ protected void getConnectionSpecificInfo( DatabaseMeta meta ) {
 
       meta.getDatabaseInterface().setNamedCluster( namedClusterList.getValue() );
     }
+
+    this.extraHandler.ifPresent( handler -> handler.getConnectionSpecificInfo( meta ) );
```
Comment (abryant-hv) [inline dbdialog/src/main/java/org/pentaho/ui/database/event/DataHandler.java:1473]: "get" usually returns something. Where does that return value go in this statement?

File: dbdialog/src/main/java/org/pentaho/ui/database/event/DbInfoHandler.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package org.pentaho.ui.database.event;
```
Comment (abryant-hv) [inline dbdialog/src/main/java/org/pentaho/ui/database/event/DbInfoHandler.java:1]: Does kettle use a top-level license file? I didn't think it did. I think we need a copywrite statement in each source file.

File: dbdialog/src/main/java/org/pentaho/ui/database/event/DbInfoHandler.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+/**
+ * A complement to {@link DataHandler} that handles
+ * database meta save/load without having to keep
+ * adding subtypes to a single class.
+ */
+public interface DbInfoHandler {
+  /** ui -> meta */
+  void getConnectionSpecificInfo( DatabaseMeta meta );
+  /** meta -> ui */
+  void setConnectionSpecificInfo( DatabaseMeta meta );
```
Comment (abryant-hv) [inline dbdialog/src/main/java/org/pentaho/ui/database/event/DbInfoHandler.java:14]: get/set seem like the wrong verbs here.

File: dbdialog/src/main/java/org/pentaho/ui/util/Launch.java [hunk 4]
```diff
@@ -32,7 +32,7 @@ public enum Status {
     Success, Failed
   }
 
-  public static Status openURL( String url ) {
```
Comment (abryant-hv) [inline dbdialog/src/main/java/org/pentaho/ui/util/Launch.java:35]: why this change?

**Review by abryant-hv — APPROVED** [review]

File: dbdialog/src/main/java/org/pentaho/ui/util/Launch.java (hunk 4 above)
Comment (tgf) [reply dbdialog/src/main/java/org/pentaho/ui/util/Launch.java:35]: to allow for overriding it in a test, it keeps opening a browser window during tests otherwise

File: dbdialog/src/main/java/org/pentaho/ui/database/event/DataHandler.java (hunk 1 above)
Comment (tgf) [reply dbdialog/src/main/java/org/pentaho/ui/database/event/DataHandler.java:1473]: just used the same method names as DataHandler has, where get/set is used for load/save. there is no return value. should change to load/save?

File: dbdialog/src/main/java/org/pentaho/ui/database/event/DbInfoHandler.java (hunk 2 above)
Comment (tgf) [reply dbdialog/src/main/java/org/pentaho/ui/database/event/DbInfoHandler.java:1]: there is a top-level license file, at least for plugins that was considered enough now

File: dbdialog/src/main/java/org/pentaho/ui/database/event/DbInfoHandler.java (hunk 3 above)
Comment (tgf) [reply dbdialog/src/main/java/org/pentaho/ui/database/event/DbInfoHandler.java:14]: was following the methods they're called in, but probably better to change to load/save

File: dbdialog/src/main/java/org/pentaho/ui/database/event/DataHandler.java (hunk 1 above)
Comment (abryant-hv) [reply dbdialog/src/main/java/org/pentaho/ui/database/event/DataHandler.java]: Eh, consistency is good.

File: dbdialog/src/main/java/org/pentaho/ui/util/Launch.java (hunk 4 above)
Comment (abryant-hv) [reply dbdialog/src/main/java/org/pentaho/ui/util/Launch.java:35]: 👍

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 1h 3m 37s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl core,dbdialog,ui
```
---
:ok_hand: All tests passed!
 
Tests run: **5326**, Failures: **0**, Skipped: **14** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9117/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: core/src/main/java/org/pentaho/di/core/database/DatabricksDatabaseMeta.java [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+
+import java.util.Optional;
+import java.util.Properties;
+
+import org.apache.commons.lang.StringUtils;
+
+/**
+ * https://docs.databricks.com/en/integrations/jdbc-odbc-bi.html#jdbc-driver
+ */
+public class DatabricksDatabaseMeta extends BaseDatabaseMeta {
```
[the remark this answers is not in the document]
Comment (tgf) [reply core/src/main/java/org/pentaho/di/core/database/DatabricksDatabaseMeta.java:14]: It is implementing `DatabaseInterface` by extending `BaseDatabaseMeta`.
Some linters scoff at reiterating implicit interface implementations, so I didn't add the `implements` here

**Review by pdesai16 — APPROVED** [review]
